### PR TITLE
Documentation on LDAP secure connection

### LINKED ISSUE #12018: Trust Store hostname-verification-policy=ANY seems to be ignored
### Describe the bug

I configured my CA's certificate according to https://www.keycloak.org/docs/latest/server_installation/#_truststore.

In docker, I mount my CA's cert using a docker volume and set X509_CA_BUNDLE to point to the cert file.

During the startup Keycloak creates its own truststore.jks file and changes standalone-ha.xml and standalone.xml accordingly. My goal, in addition to that, is to ignore any errors regarding the certificate names or SANs, because I have to work with an Active Directory  CA which is untrusted outside of itself and I cannot control the certificates themselves.

In order to achieve that I created a cli script that also sets hostname-verification-policy to ANY during startup.

This is the resulting XML:

```xml
<spi name="truststore">
    <provider name="file" enabled="true">
        <properties>
            <property name="file" value="/opt/jboss/keycloak/standalone/configuration/keystores/truststore.jks"/>
            <property name="password" value="C6ZJ+p3zj+7Bs6OzGNk1yN039CHoTOrV2P6YiBSDUG4="/>
            <property name="hostname-verification-policy" value="ANY"/>
            <property name="disabled" value="false"/>
        </properties>
    </provider>
</spi>
```

Unfortunately, that does not work at all:

```
10:18:23,822 DEBUG [org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager] (default task-4) Creating LdapContext using properties: [{java.naming.security.authentication=simple, java.naming.security.principal=CN=Kessler,CN=Users,DC=wghoch4,DC=local, java.naming.factory.initial=com.sun.jndi.ldap.LdapCtxFactory, java.naming.ldap.factory.socket=org.keycloak.truststore.SSLSocketFactory, java.naming.security.credentials=**************************************, java.naming.provider.url=ldaps://arwen.home.therightstuff.de:636}]
10:18:23,827 ERROR [org.keycloak.services] (default task-4) KC-SERVICES0055: Error when authenticating to LDAP: simple bind failed: arwen.home.therightstuff.de:636: javax.naming.CommunicationException: simple bind failed: arwen.home.therightstuff.de:636 [Root exception is javax.net.ssl.SSLHandshakeException: No subject alternative DNS name matching arwen.home.therightstuff.de found.]
```

### Version

15.0.2

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by hmlnarik**
Closes: #12018

### CODE REVIEW

File: docs/guides/server/keycloak-truststore.adoc [hunk 1]
```diff
@@ -38,6 +38,8 @@ For HTTPS requests, this option verifies the hostname of the server's certificat
 * `ANY` means that the hostname is not verified.
 * `WILDCARD` allows wildcards in subdomain names, such as *.foo.com.
 * When using `STRICT`, the Common Name (CN) must match the hostname exactly.
++
+Please note that this settings does not apply to LDAP secure connections which requires strict hostname checking.
```
Comment (pedroigor) [inline docs/guides/server/keycloak-truststore.adoc:42]: In the `file` property documentation, does it make sense to replace `HTTPS` with `TLS`? The trust store is not only used in LDAP, as you are making clear here.

I think the beginning of this guide is already generic. But perhaps you can spot anything.

Comment (hmlnarik) [reply docs/guides/server/keycloak-truststore.adoc:42]: Thank you, fixed

Comment (andymunro) [inline docs/guides/server/keycloak-truststore.adoc:42]:
```suggestion
Please note that this setting does not apply to LDAP secure connections, which require strict hostname checking.
```

Comment (andymunro) [reply docs/guides/server/keycloak-truststore.adoc:42]: BTW, a subordinate clause that begins with "which" begins with a comma.

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good, @hmlnarik. One minor change.

File: docs/guides/server/keycloak-truststore.adoc (hunk 1 above)
Comment (hmlnarik) [reply docs/guides/server/keycloak-truststore.adoc]: Thank you!

**Review by pedroigor — APPROVED** [review]

**Pull request merged**